contracts/manager/BaseManagerV2.sol

```diff
@@ -0,0 +1,596 @@
+     * methodologist needs to explicitly approve protection arrangements under those conditions.
+     *
+     * NOTE: If adding a fee extension while protecting a fee module, it's important to set the
+     * module `feeRecipient` to the new extension's address (ideally before this call).
+     *
+     * @param  _module          Module to protect
+     * @param  _extensions        Array of extensions to authorize for protected module
+     */
+    function protectModule(address _module, address[] memory _extensions)
+        external
+        upgradesPermitted
+        onlyOperator
+    {
+        require(setToken.getModules().contains(_module), "Module not added yet");
+        _protectModule(_module, _extensions);
+
+        emit ModuleProtected(_module, _extensions);
+    }
+
+    /**
+     * METHODOLOGIST ONLY: Marks a currently protected module as unprotected and deletes its authorized
+     * extension registries. Removes old module from the protected modules list.
+     *
+     * Called by the methodologist when they want to cede control over a protected module without triggering
+     * an emergency (for example, to remove it because its dead).
+     *
+     * @param  _module          Module to revoke protections for
+     */
+    function unProtectModule(address _module) external onlyMethodologist {
+        _unProtectModule(_module);
+
+        emit ModuleUnprotected(_module);
+    }
+
+    /**
+     * MUTUAL UPGRADE: Replaces a protected module. Operator and Methodologist must each call this
+     * function to execute the update.
+     *
+     * > Marks a currently protected module as unprotected
+     * > Deletes its authorized extension registries.
+     * > Removes old module from SetToken.
+     * > Adds new module to SetToken.
+     * > Marks `_newModule` as protected and authorizes new extensions for it.
+     *
+     * Used when methodologists wants to guarantee that an existing protection arrangement is replaced
+     * with a suitable substitute (ex: upgrading a StreamingFeeSplitExtension).
+     *
+     * NOTE: If replacing a fee module, it's necessary to set the module `feeRecipient` to be
+     * the new fee extension address after this call. Any fees remaining in the old module's
+     * de-authorized extensions can be distributed by calling `distribute()` on the old extension.
+     *
+     * @param _oldModule        Module to remove
+     * @param _newModule        Module to add in place of removed module
+     * @param _newExtensions      Extensions to authorize for new module
+     */
+    function replaceProtectedModule(address _oldModule, address _newModule, address[] memory _newExtensions)
+        external
+        mutualUpgrade(operator, methodologist)
+    {
+        _unProtectModule(_oldModule);
+
+        setToken.removeModule(_oldModule);
+        setToken.addModule(_newModule);
+
+        _protectModule(_newModule, _newExtensions);
+
+        ReplacedProtectedModule(_oldModule, _newModule, _newExtensions);
```

Comment:
Do we not need the `emit` here?

Reply:
Apparently it just works. However added the missing ones for consistency's sake in 87c8fcf